ast/src/analyzed/mod.rs

```diff
@@ -969,8 +969,7 @@ impl<T> SelectedExpressions<T> {
     /// @returns true if the expression contains a reference to a next value of a
     /// (witness or fixed) column
     pub fn contains_next_ref(&self) -> bool {
-        self.selector
```

Comment:
What about `self.children().any(|e| e.contains_next_ref())`?

Reply:
I mean (not as part of this PR), but we could also get rid of the `contains_next_ref` functions above Expression and just work with `all_children()` called on an AST node.

Reply:
Changed to use children